ops/pebble.py

```diff
@@ -1338,6 +1338,11 @@ def wait_output(self) -> Tuple[AnyStr, Optional[AnyStr]]:
             out = io.BytesIO()
             err = io.BytesIO() if self.stderr is not None else None
 
+        if self.stdout is None:
+            raise TypeError(
+                "stdout is already being consumed by another object provided to the exec function"
```

Comment:
That said, two nits:

1) Could we move this up to the top of the function, so that we're checking inputs before performing any logic / allocations?
2) Perhaps we can slightly improve the error message to make it easier for the caller to know what (not) to do. Maybe "can't use wait_output() when exec was called with the stdout argument; use wait() instead".

Reply:
Okay, updated, thanks!